rocAL/rocAL_pybind/amd/rali/pipeline.py

```diff
@@ -176,47 +184,48 @@ def process_calls(self, output_image):
     def build(self):
         """Build the pipeline using raliVerify call
         """
-        outputs = self.define_graph()
-        self.process_calls(outputs[0])
-
-        #Checks for Casting "Labels" as last node & Box Encoder as last Prev node
-        if(len(outputs)==3):
-            if((isinstance(outputs[1],list)== False) & (isinstance(outputs[2],list)== False)):
-                if((outputs[2].prev is not None) | (outputs[1].prev is not None)):
-                    if(outputs[2].prev.data == "Cast") :
-                        self._castLabels = True
-                        if(outputs[2].prev.prev.prev.data is not None):
-                            if(outputs[2].prev.prev.prev.data == "BoxEncoder"):
+        if self.define_graph()!=0:
+            outputs = self.define_graph()
+            self.process_calls(outputs[0])
+
+            #Checks for Casting "Labels" as last node & Box Encoder as last Prev node
+            if(len(outputs)==3):
+                if((isinstance(outputs[1],list)== False) & (isinstance(outputs[2],list)== False)):
+                    if((outputs[2].prev is not None) | (outputs[1].prev is not None)):
+                        if(outputs[2].prev.data == "Cast") :
+                            self._castLabels = True
+                            if(outputs[2].prev.prev.prev.data is not None):
+                                if(outputs[2].prev.prev.prev.data == "BoxEncoder"):
```

Comment:
In the next version, BoxEncoder needs to be function with separate node in rocal